From: Rene Kita <mail@rkta.de>
Subject: Remove left-over FORMAT_NICE code

The configure option was removed in
commit 2ba7abd687a38b40b6d1d2c62c0e432a5b9775a5
Author: Fumitoshi UKAI <ukai@debian.or.jp>
Date:   2003-03-10T17:28:52+00:00

    [w3m-dev 03804] fix autoconf
    * acinclude.m4 (AC_W3M_MODEL): deleted
            (AC_W3M_COLOR): default enable
            (AC_W3M_ANSI_COLOR): default enable
            (AC_W3M_BG_COLOR): default enable
            (AC_W3M_SHOW_PARAMS): deleted
            (AC_W3M_DICT): default enable
            (AC_W3M_HISTORY): default enable
            (AC_W3M_VIEW_UNSEENOBJECTS): deleted
            (AC_W3M_FORMAT_NICE): deleted
            (AC_W3M_ID_EXT): deleted
            (AC_W3M_FTPPASS_HOSTNAMEGEN): deleted
            (AC_W3M_NNTP): default enabled
            (AC_W3M_LANG): autodetect whether japanese is enabled
            (AC_W3M_KANJI_SYMBOLS): default enabled
            (AC_W3M_EMACS_LIKE_LINEEDIT): deleted
            (AC_W3M_VI_PREC_NUM): deleted
            (AC_W3M_MARK): deleted
            (AC_W3M_LABEL_TOPLINE): deleted
            (AC_W3M_NEXTPAGE_TOPLINE): deleted
            (AC_W3M_SSL): fix without ssl
            (AC_W3M_IMAGE): default enabled
            (AC_W3M_XFACE): check uncompface
            (AC_W3M_IPv6): enable_ipv6 if inet6 lib found
    * configure.in (AC_PROG_CPP): added
    From: Fumitoshi UKAI  <ukai@debian.or.jp>
---
 config.h.dist |    1 -
 config.h.in   |    1 -
 file.c        |   57 ---------------------------------------------------------
 fm.h          |    3 ---
 4 files changed, 62 deletions(-)

--- a/config.h.in
+++ b/config.h.in
@@ -81,7 +81,6 @@
 #undef USE_XFACE
 #undef USE_DICT
 #undef USE_HISTORY
-#undef FORMAT_NICE
 #define ID_EXT
 #undef CLEAR_BUF
 #undef INET6
--- a/file.c
+++ b/file.c
@@ -2490,9 +2490,6 @@ set_breakpoint(struct readbuffer *obuf,
     obuf->bp.pos = obuf->pos;
     obuf->bp.tlen = tag_length;
     obuf->bp.flag = obuf->flag;
-#ifdef FORMAT_NICE
-    obuf->bp.flag &= ~RB_FILL;
-#endif				/* FORMAT_NICE */
     obuf->bp.top_margin = obuf->top_margin;
     obuf->bp.bottom_margin = obuf->bottom_margin;
 
@@ -2897,47 +2894,6 @@ flushline(struct html_feed_environ *h_en
 	else if (RB_GET_ALIGN(obuf) == RB_LEFT && obuf->flag & RB_INTABLE) {
 	    align(lbuf, width, ALIGN_LEFT);
 	}
-#ifdef FORMAT_NICE
-	else if (obuf->flag & RB_FILL) {
-	    char *p;
-	    int rest, rrest;
-	    int nspace, d, i;
-
-	    rest = width - get_Str_strwidth(line);
-	    if (rest > 1) {
-		nspace = 0;
-		for (p = line->ptr + indent; *p; p++) {
-		    if (*p == ' ')
-			nspace++;
-		}
-		if (nspace > 0) {
-		    int indent_here = 0;
-		    d = rest / nspace;
-		    p = line->ptr;
-		    while (IS_SPACE(*p)) {
-			p++;
-			indent_here++;
-		    }
-		    rrest = rest - d * nspace;
-		    line = Strnew_size(width + 1);
-		    for (i = 0; i < indent_here; i++)
-			Strcat_char(line, ' ');
-		    for (; *p; p++) {
-			Strcat_char(line, *p);
-			if (*p == ' ') {
-			    for (i = 0; i < d; i++)
-				Strcat_char(line, ' ');
-			    if (rrest > 0) {
-				Strcat_char(line, ' ');
-				rrest--;
-			    }
-			}
-		    }
-		    lbuf = newTextLine(line, width);
-		}
-	    }
-	}
-#endif				/* FORMAT_NICE */
 #ifdef TABLE_DEBUG
 	if (w3m_debug) {
 	    FILE *f = fopen("zzzproc1", "a");
@@ -6755,15 +6711,8 @@ HTMLlineproc0(char *line, struct html_fe
 		    bp = obuf->line->ptr + obuf->bp.len;
 		    line = Strnew_charp(bp);
 		    Strshrink(obuf->line, obuf->line->length - obuf->bp.len);
-#ifdef FORMAT_NICE
-		    if (obuf->pos - i > h_env->limit)
-			obuf->flag |= RB_FILL;
-#endif				/* FORMAT_NICE */
 		    back_to_breakpoint(obuf);
 		    flushline(h_env, obuf, indent, 0, h_env->limit);
-#ifdef FORMAT_NICE
-		    obuf->flag &= ~RB_FILL;
-#endif				/* FORMAT_NICE */
 		    HTMLlineproc1(line->ptr, h_env);
 		}
 	    }
@@ -6784,13 +6733,7 @@ HTMLlineproc0(char *line, struct html_fe
 	    i = 1;
 	indent = h_env->envs[h_env->envc].indent;
 	if (obuf->pos - i > h_env->limit) {
-#ifdef FORMAT_NICE
-	    obuf->flag |= RB_FILL;
-#endif				/* FORMAT_NICE */
 	    flushline(h_env, obuf, indent, 0, h_env->limit);
-#ifdef FORMAT_NICE
-	    obuf->flag &= ~RB_FILL;
-#endif				/* FORMAT_NICE */
 	}
     }
 }
--- a/fm.h
+++ b/fm.h
@@ -668,9 +668,6 @@ struct readbuffer {
 #define RB_SPECIAL	(RB_PRE | RB_PRE_INT | RB_SCRIPT | RB_STYLE | RB_PLAIN | RB_NOBR)
 #define RB_PLAIN_PRE	0x40000
 
-#ifdef FORMAT_NICE
-#define RB_FILL		0x80000
-#endif				/* FORMAT_NICE */
 #define RB_DEL		0x100000
 #define RB_S		0x200000
 #define RB_HTML5	0x400000
--- a/config.h.dist
+++ b/config.h.dist
@@ -27,7 +27,6 @@
 /*
  * Do word fill
  */
-#undef FORMAT_NICE
 
 /*
  * Enable id attribute
